src/helpers/template-helper.js

```diff
@@ -22,15 +22,66 @@ const templateEngine = new Liquid()
 
 /**
  * Add filter findAgent to template engine.
- * Syntaxe  {{ microservice findAgent: agents }}
+ * Syntaxe  {{ microservice | findMicroserviceAgent }}
  */
-templateEngine.registerFilter('findAgent', (microservice, agents) => {
-  const targetAgent = agents ? agents.find(agent => agent.uuid === microservice.iofogUuid) : undefined
-  return targetAgent
-})
 
-function findEdgeResourcehandler (name, version) {
-  const result = this.context.environments._user ? EdgeResourceService.getEdgeResource({ name, version }, this.context.environments._user) : Promise.resolve(undefined)
+function findMicroserviceAgentHandler (microservice) {
+  const user = this.context.environments._user
+  if (!user) {
+    return undefined
+  }
+  const result = FogService.getFogEndPoint({ uuid: microservice.iofogUuid }, user, false)
+  return result
+}
+
+async function findEdgeResourcehandler (name, version) {
+  const key = `${name}/${version}`

```

Comment:
`version` can be undifined if searching without version number. And caching works also in this case